packages/flame/lib/src/gestures/events.dart

```diff
@@ -0,0 +1,179 @@
+import 'package:flutter/gestures.dart';
+
+import '../../extensions.dart';
+
+abstract class BaseInfo<T> {
+  final T raw;
+
+  BaseInfo(this.raw);
+}
+
+abstract class PositionInfo<T> extends BaseInfo<T> {
+  final Vector2 position;
+  final Vector2 gameWidgetPosition;
+  final Vector2 globalPosition;
+
+  PositionInfo(
+    this.position,
+    this.gameWidgetPosition,
+    this.globalPosition,
+    T raw,
+  ) : super(raw);
+}
+
+class TapDownInfo extends PositionInfo<TapDownDetails> {
+  TapDownInfo(
+    Vector2 position,
+    Vector2 gameWidgetPosition,
+    Vector2 globalPosition,
+    TapDownDetails raw,
+  ) : super(position, gameWidgetPosition, globalPosition, raw);
+}
+
+class TapUpInfo extends PositionInfo<TapUpDetails> {
+  TapUpInfo(
+    Vector2 position,
+    Vector2 gameWidgetPosition,
+    Vector2 globalPosition,
+    TapUpDetails raw,
+  ) : super(position, gameWidgetPosition, globalPosition, raw);
+}
+
+class LongPressStartInfo extends PositionInfo<LongPressStartDetails> {
+  LongPressStartInfo(
+    Vector2 position,
+    Vector2 gameWidgetPosition,
+    Vector2 globalPosition,
+    LongPressStartDetails raw,
+  ) : super(position, gameWidgetPosition, globalPosition, raw);
+}
+
+class LongPressEndInfo extends PositionInfo<LongPressEndDetails> {
+  final Vector2 velocity;
+
+  LongPressEndInfo(
+    Vector2 position,
+    Vector2 gameWidgetPosition,
+    Vector2 globalPosition,
+    this.velocity,
+    LongPressEndDetails raw,
+  ) : super(position, gameWidgetPosition, globalPosition, raw);
+}
+
+class LongPressMoveUpdateInfo extends PositionInfo<LongPressMoveUpdateDetails> {
+  LongPressMoveUpdateInfo(
+    Vector2 position,
+    Vector2 gameWidgetPosition,
+    Vector2 globalPosition,
+    LongPressMoveUpdateDetails raw,
+  ) : super(position, gameWidgetPosition, globalPosition, raw);
+}
+
+class ForcePressInfo extends PositionInfo<ForcePressDetails> {
+  final double pressure;
+
+  ForcePressInfo(
+    Vector2 position,
+    Vector2 gameWidgetPosition,
+    Vector2 globalPosition,
+    this.pressure,
+    ForcePressDetails raw,
+  ) : super(position, gameWidgetPosition, globalPosition, raw);
+}
+
+// TODO(erick) This class, and the next one has a bunch of info, we need to check the ones
+// that could wrapped directly by the info class
+class PointerScrollInfo extends BaseInfo<PointerScrollEvent> {
+  final Vector2 position;
+  final Vector2 gameWidgetPosition;
+  final Vector2 scrollDelta;
+
```

Comment:
As I see from `_parsePointerScrollEvent`, these are values totally dependant on what is store on `raw`. 
Given that, it may be misleading to keep this value as described in these classes, since semantically they should not diverge from the fields on the event class. 

There is a logic for  `scrollDelta` for example, it should be `event.scrollDelta` as Vector2.
That logic is not here, therefore this class is not self-contained and its very principle of existence is totally controlled by external elements. 

As an alternative to this, we can have this information as either _getters_ or _late finals_.

Example with late finals:
```dart
late final scrollDelta = raw.scrollDelta.toVector2()
```

And then the constructor would only receive `raw`, delegating less job to a random private method somewhere in the engine. Also, it would be more clear for the engine users seeing this class to understand how these values are calculated and where they come from. 

I think the same applies for every `*Info` class here.

Reply:
yeah, that "stumbles" on another suggestion that @spydon gave that follows the same idea to make these all late getters.

Gonna do that in the sequence.